### ISSUE TITLE
feature request: directly booting into any installed game (also non-steam games!)

### ISSUE THREAD

**Issue by stapelberg** [ValveSoftware/SteamOS#432]
I’m replacing my [OpenELEC](http://openelec.tv/)-based media center with a steam machine so that I can both consume media and play games.

In order to keep the setup as simple as possible for the most common use-case (consuming media), I’d like to reduce the number of actions necessary to get to the media as far as possible.

I’ve installed [kodi](http://kodi.tv/) (a media center application) and added it as a non-steam shortcut via “Settings” → “Add Library Shortcut”.

By default, when booting into SteamOS, one needs to select “Library”, then “Installed”, then “Kodi”, then “Your Shortcut”, then “PLAY”. That’s 5+ interactions which I’d like to skip.

I can achieve what I want by replacing the last line of `/usr/bin/steamos-session` with:
```
steam -tenfoot -steamos -enableremotecontrol steam://rungameid/11312753474682224640
```

However, figuring out the game id for a non-steam shortcut is really hard for regular users, and modifying files of the operating system is not something I’d like to keep doing.

Hence, could we please get a new setting, where we can chose which game should be ran at startup?

Thanks!

**Comment by ProfessorKaos64** [comment]
This isn't a SteamOS issue, The step you have provided isn't commonplace, and I doubt will ever be accepted. SteamOS needs to boot into steamos-session / steamcompmgr cleanly.

**Comment by stapelberg** [comment]
> This isn't a SteamOS issue,

If it’s not related to SteamOS, to which part of the steam machine is it related to?

> SteamOS needs to boot into steamos-session / steamcompmgr cleanly.

I don’t understand what you mean, can you explain please? AIUI, with my modification, SteamOS is launching steamos-session, which launches steam and proceeds to run a specific app. What part of this is not clean?

**Comment by ProfessorKaos64** [comment]
I honestly don't think Valve would allow such a modification as a setting that would not boot to the "home screen" of SteamOS BPM, the intended experience. While this is something nice for tinkerers, like myself, I don't see this ever becoming a standard feature. Who knows, just my 2 cents. I really don't think it's painstaking to click my library and start Kodi.

**Comment by jdunne525** [comment]
I agree this feature request would be very helpful.  I've been trying to figure out how to do this for the past 2 days.  

Stapelberg: how did you figure out the gameid for a non-steam game?

In the family room, most of the time my HTPC system is used for playing movies and such via Kodi.  It is pretty rare that I'll play games, and as such booting directly into kodi is better.  Also, my wife and kids use Kodi and I don't want to complicate things for them by adding extra steps.

**Comment by stapelberg** [comment]
@jdunne525 I’ve used a modified version of https://github.com/boppreh/steamgrid/blob/master/games.go to generate the gameid. I agree that this is much too complicated, and would really appreciate it if SteamOS would just introduce the functionality we’re looking for :).

**Comment by jdunne525** [comment]
I gave up on Steamos.  I switched to Kodibuntu and installed steam on that.  It works much smoother, though I will admit that was a bit painful also getting sound to work through my HDMI setup.  Kodi wasn't running well from Steamos for me.
